packages/util/src/__specs__/polymorphism.spec.tsx

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+
+import { render } from '@testing-library/react'
+import React, { MouseEvent } from 'react'
+
+import { forwardRefWithAs, forwardRefWithAsAndStatics } from '..'
+
+interface Props {
+  optional?: boolean
+}
+const Button = forwardRefWithAs<Props, 'button'>((props, ref) => {
+  const { as: Comp = 'button', optional, ...rest } = props
+  return <Comp data-optional={optional} ref={ref} {...rest} />
+})
+
+const Anchor = forwardRefWithAs<Props, 'a'>((props, ref) => {
+  const { as: Comp = 'a', optional, ...rest } = props
+  return <Comp data-optional={optional} ref={ref} {...rest} />
+})
+
+interface Statics {
+  Anchor: typeof Anchor
+  Button: typeof Button
+}
+
+const CompoundComp = forwardRefWithAsAndStatics<Props, 'div', Statics>(
+  (props, ref) => {
+    const { as: Comp = 'div', optional, ...rest } = props
+
+    return <Comp data-optional={optional} ref={ref} {...rest} />
+  }
+)
+CompoundComp.Anchor = Anchor
+CompoundComp.Button = Button
+
+describe('polymorphism', () => {
+  describe('#forwardRefWithAs', () => {
+    it('renders without type errors', () => {
+      expect.assertions(0)
+
+      render(
+        <>
+          {/* Anchor accepts optional prop */}
```

Comment:
Idea: Could write as separate `it('does a thing', () => {})` calls and avoid the comments.

Reply:
yeah i can look into that